src/Raven.Server/Documents/ETL/Providers/Elasticsearch/ElasticSearchEtl.cs

```diff
@@ -104,47 +106,49 @@ protected override int LoadInternal(IEnumerable<ElasticSearchIndexWithRecords> r
 
             _client ??= ElasticSearchHelper.CreateClient(Configuration.Connection);
 
-            foreach (var index in records)
+            using (((BlittableJsonElasticSerializer)_client.SourceSerializer).SetContext(context))
             {
-                string indexName = index.IndexName.ToLower();
-
-                EnsureIndexExistsAndValidateIfNeeded(indexName, index);
-
-                if (index.InsertOnlyMode == false)
-                    count += DeleteByQueryOnIndexIdProperty(index);
-
-                if (index.Inserts.Count > 0)
+                foreach (var index in records)
                 {
-                    var streamHandler = PostData.StreamHandler(index.Inserts, (inserts, stream) =>
+                    string indexName = index.IndexName.ToLower();
+    
+                    EnsureIndexExistsAndValidateIfNeeded(indexName, index);
+    
+                    if (index.InsertOnlyMode == false)
+                        count += DeleteByQueryOnIndexIdProperty(index);
+    
+                    if (index.Inserts.Count == 0)
                     {
-                        foreach (ElasticSearchItem insert in inserts)
-                        {
-                            if (insert.TransformationResult == null)
-                                continue;
-
-                            stream.Write(IndexBulkActionBytes);
-
-                            using (var json = EnsureLowerCasedIndexIdProperty(context, insert.TransformationResult, index))
-                            using (var writer = new BlittableJsonTextWriter(context, stream))
-                            {
-                                writer.WriteNewLine();
-                                writer.WriteObject(json);
-                                writer.WriteNewLine();
-                            }
-
-                            count++;
-                        }
-                    }, (i, s, token) => throw new NotSupportedException("We don't use async bulk method"));
-
+                        continue; // we avoid requesting bulk without body (with no create clauses), it causes an error  
+                    }
+    
+                    var bulkRequestDescriptor = new BulkRequestDescriptor().Index(indexName).Refresh(Elastic.Clients.Elasticsearch.Refresh.WaitFor);
+                    var toDispose = new List<BlittableJsonReaderObject>();
+                    foreach (var insert in index.Inserts.Where(insert => insert.TransformationResult != null))
```

Comment:
Instead of using Linq `.Where(insert => insert.TransformationResult != null)` let's convert it to previous code:

```
if (insert.TransformationResult == null)
                                continue;
```

Linq makes additional allocations. We can easily avoid that

Reply:
done